libbeat/monitoring/report/report.go

```diff
@@ -0,0 +1,127 @@
+package report
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/publisher"
+)
+
+type config struct {
+	// allow for maximum one reporter being configured
+	Reporter common.ConfigNamespace `config:",inline"`
+}
+
+type Reporter interface {
```

Comment:
nice thing about interfaces is, if they are assignable (same definition or sub-type relation), they can be used interchangeable. I'm aware of us having a mix of this interface.

Besides similar definition, giving it a name fitting the packages usage is sometimes nice to have. Plus, the Reporter interface might change. I'm not too much a fan of the 'Start(), Stop()` interfaces, as this clearly mandates/leaks usage of go-routines and very often channels for communication... I think in 'Reporter' we can get rid of 'Start' :)